WordPressKit/Jetpack Scan/JetpackScanThreat.swift

```diff
@@ -0,0 +1,294 @@
+import Foundation
+
+public struct JetpackScanThreat: Decodable {
+    public enum ThreatStatus: String {
+        case fixed
+        case ignored
+        case current
+
+        case unknown
+    }
+
+    public enum ThreatType: String {
+        case core
+        case file
+        case plugin
+        case theme
+        case database
+
+        case unknown
+
+        init(threat: JetpackScanThreat) {
+            // Logic used from https://github.com/Automattic/wp-calypso/blob/5a6b257ad97b361fa6f6a6e496cbfc0ef952b921/client/components/jetpack/threat-item/utils.ts#L11
+            if threat.diff != nil {
+                self = .core
+            } else if threat.context != nil {
+                self =  .file
+            } else if let ext = threat.extension {
+                self = ThreatType(rawValue: ext.type.rawValue) ?? .unknown
+            } else if threat.rows != nil || threat.signature.contains(Constants.databaseSignature) {
+                self = .database
+            } else {
+                self = .unknown
+            }
+        }
+
+        private struct Constants {
+            static let databaseSignature = "Suspicious.Links"
+        }
+    }
+
+    /// The ID of the threat
+    public let id: Int
+
+    /// The name of the threat signature
+    public let signature: String
+
+    /// The description of the threat signature
+    public let description: String
+
+    /// The date the threat was first detected
+    public let firstDetected: Date
+
+    /// Whether the threat can be automatically fixed
+    public let fixable: JetpackScanThreatFixer?
+
+    /// The filename
+    public let fileName: String?
+
+    /// The status of the threat (fixed, ignored, current)
+    public let status: ThreatStatus?
+
+    /// The date the threat was fixed on
+    public let fixedOn: Date?
+
+    /// More information if the threat is a extension type (plugin or theme)
+    public let `extension`: JetpackThreatExtension?
+
+    /// The type of threat this is
+    private(set) var type: ThreatType
+
+    /// If this is a file based threat this will provide code context to be displayed
+    /// Context example:
+    /// 3: start test
+    /// 4: VIRUS_SIG
+    /// 5: end test
+    /// marks: 4: (0, 9)
+    public let context: JetpackThreatContext?
+
+    // Core modification threats will contain a git diff string
+    public let diff: String?
+
+    // Database threats will contain row information
+    public let rows: [String: Any]?
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        // Define a placeholder type
+        type = .unknown
```

Comment:
So I can't remove this one, or else the the last line `type = ThreatType(threat: self)` won't work since I need to pass self to it, but it will throw `Variable 'self.type' used before being initialized`.

I tried working around it but wasn't able to. Any ideas?